packages/vite/src/node/build.ts

```diff
@@ -546,8 +546,8 @@ export async function build(
     input,
     plugins,
     external,
-    onwarn(warning, warn) {
-      onRollupWarning(warning, warn, config)
+    onLog(level, log, defaultHandler) {
+      onRollupLog(level, log, defaultHandler, config)
```

Comment:
Would you check if `onwarn` is called before `onLog` or not? If `onLog` is called before `onwarn`, I think we'll need to pass both `onwarn` and `onLog`.

Reply:
@sapphi-red 
Thank you for your review!

When rollup calls the logger with warn, `onwarn` is called after `onLog`. And my understanding is that `onwarn` is only called when the default handler is called at `onLog`.

In the event of a warning, only logs that are not handled by the handling implemented in `onRollupLog` (ignoring specific warning logs or handling plugin logs) will be passed to the default handler and `onwarn` will be called. On that premise, passing the same implementation to `onLog` and `onwarn` does not make sense because the same handling (which is also executed in onlog) is executed.

What do you think about this?

Reply:
Sorry that it took time, thanks for checking it :pray: